src/AppBundle/Repository/PostRepository.php

```diff
@@ -38,4 +38,24 @@ public function findLatest($limit = Post::NUM_ITEMS)
             ->getResult()
         ;
     }
+
+    public function findByTerms(array $terms, $limit = Post::NUM_ITEMS)
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
```

Comment:
It's not a Symfony best practice, it's a PHP one. Having `$em` as a variable is more a legacy from Doctrine than a Symfony convention.